src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,189 @@
+        # Set Grants 
+        grant_a['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3620000000.0,
+            'highFrequency': 3630000000.0
+        }
+        grant_c['operationParam']['operationFrequencyRange'] = {
+            'lowFrequency': 3640000000.0,
+            'highFrequency': 3650000000.0
+        }
+        grant_request = [grant_a, grant_c]
+        request = {'grantRequest': grant_request}     
+        # Send grant requests
+        grant_response = self._sas.Grant(request)['grantResponse']
+        grant_ids = []
+        for resp in grant_response:
+            self.assertEqual(resp['response']['responseCode'], 0)
+            grant_ids.append(resp['grantId'])
+        # STEP 2
+        # Inject PPA
+        pal_low_frequency = 3550000000.0
+        pal_high_frequency = 3560000000.0
+        pal_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'pal_record_0.json')))
+        ppa_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+        ppa_record, pal_record = makePpaAndPalRecordsConsistent(ppa_record,
+                                                                [pal_record],
+                                                                pal_low_frequency,
+                                                                pal_high_frequency,
+                                                                device_a['userId'])
+        self._sas_admin.InjectPalDatabaseRecord(pal_record[0])
+        self._sas_admin.InjectZoneData({'record': ppa_record})
+        # Inject Esc Sensor
+        esc_sensor_record = json.load(
+          open(os.path.join('testcases', 'testdata', 'esc_sensor_record_0.json')))
+        self._sas_admin.InjectEscSensorDataRecord({'record': esc_sensor_record})
+        # STEP 3
+        self._sas_admin.TriggerFullActivityDump()
+        # STEP 4
+        response = self._sas.GetFullActivityDump()
+        # STEP 5
+        self.assertContainsRequiredFields("FullActivityDump.schema.json", response)
+        # STEP 6 AND CHECK
+        self.assertGreaterEqual(3, len(response['files']))
+        self.assertLessEqual(2 + len(cbsd_ids), len(response['files']))
+        cbsd_dump_files = [dump_file for dump_file in response['files'] if dump_file['recordType'] ==  'cbsd']
+        esc_sensor_dump_file = [dump_file for dump_file in response['files'] if dump_file['recordType'] ==  'esc_sensor']
+        zone_dump_file = [dump_file for dump_file in response['files'] if dump_file['recordType'] ==  'zone']        
+        # Verify the record type
+        # Verify the response files with ActivityDumpFile.schema.json Object schema
+        for dump_file in [esc_sensor_dump_file, zone_dump_file]:
+            self.assertContainsRequiredFields("ActivityDumpFile.schema.json",
+                                               dump_file[0])
+            self.assertEqual(1, len(dump_file))
```

Comment:
I don't think that this will work once we generalize the test to use N2 > 0 PPAs and N3 > 0 ESC sensors. You could leave it for now but it will only need to be removed or changed in the next version. The same goes for lines 176-189 below: they work for now but will fail in the general case.

That said, perhaps the best strategy is to mark them as needing update but wait to make the changes in the PR which makes the test case configurable.